src/os/bluestore/ZonedAllocator.cc

```diff
@@ -22,7 +22,7 @@ ZonedAllocator::ZonedAllocator(CephContext* cct,
 			       int64_t size,
 			       int64_t blk_size,
 			       std::string_view name)
-    : Allocator(name, size, blk_size),
+    : Allocator(name, size, blk_size & 0x00000000ffffffff),
```

Comment:
would be better if we can have a function or a macro wrapping this logic of 
```
      // To avoid interface changes, we piggyback zone size and the first
      // sequential zone number onto the first 32 bits of 64-bit |blk_size|.
      // The last 32 bits of |blk_size| is holding the actual block size.
```

but apparently, this is not in the scope of this change..

Reply:
Added to a todo list.